scope_extensions-explainer.md

```diff
@@ -0,0 +1,104 @@
+# Extended scopes for Web Apps
+
+## Overview
+
+This document describes a new `scope_extensions` manifest member that enables
+web apps to extend their
+[scope](https://www.w3.org/TR/appmanifest/#understanding-scope) to other
+origins.
+
+## Use Cases / Goals
+
+- Allow sites that control multiple subdomains and top level domains to behave
+  as one contiguous web app.\
+  E.g. a site may span `example.com`, `example.co.uk` and `support.example.com`.
+
+- Allow web apps to capture user navigations to sites they are affiliated with.\
+  E.g. "News Aggregator App" capturing links navigations to examplenewssite.com.
+
+## Background
+
+Web app scope (defined by the `scope` field) is currently used for:
+1. Constraining urls like start_url, file handlers, share target.
+1. Defining the set of link navigations that can be captured by the app.
+1. Determining whether an app window's root document has left the app's scope
+   (possibly invoking window UI informing the user of this).
+
+The `scope_extensions` mechanism can expand all these behaviours to include
+other origins given agreement between the web app's primary origin and the
+associated origins.
+
+## Proposal
+
+1. Add a `scope_extensions` member to the web app manifest specifying a list of
+   origin patterns to associate with.
+
+   Example manifest located at `https://example.com/manifest.webmanifest`:
+   ```
+   {
+     "id": "",
+     "name": "Example",
+     "display": "standalone",
+     "start_url": "/index.html",
+     "scope_extensions": [
+       {"origin": "*.example.com"},
+       {"origin": "example.co.uk"},
+       {"origin": "*.example.co.uk"}
+     ]
+   }
+   ```
+   In this example the Example app is extending its app scope to all its
+   subdomains along with its .co.uk site and its subdomains.
+
+1. Specify a `web-app-origin-association.json` file format that must be located
```

Comment:
This should probably be aligned with https://web.dev/pwa-url-handler/#the-web-app-origin-association-file:

```json
{
  "web_apps": [
    {
      "manifest": "https://music.example.com/manifest.json",
      "details": {
        "paths": ["/*"],
        "exclude_paths": ["/internal/*"]
      }
    }
  ]
}
```

Reply:
Using the manifest as an identifier isn't compatible with where web app identifiers are headed. I figured if that needed changing then everything is up for grabs to change as well but don't feel strongly about this.

Reply:
Context: https://github.com/philloooo/pwa-unique-id/blob/main/explainer.md#recommended-1-start_url_origin--specified-id

Reply:
Aware of the context. Just if the file is named the same, it shouldn't be incompatible with the other proposal (well noting that everythig, this and the other proposal, is still in flux). Maybe just add a remark that calls the reader's attention to the other proposal, as they [have done](https://github.com/WICG/pwa-url-handler/blob/main/explainer.md#:~:text=Web%20apps%20can%20be%20identified%20by%20their%20manifest%20URL%20currently%20before%20a%20unique%20identifier%20is%20standardized.).

Reply:
I'm hoping we can avoid maintaining backwards compatibility with something that hasn't shipped to the general web.

Adding context SGTM, will update.

Reply:
Done.